apps/files_sharing/js/app.js

```diff
@@ -214,16 +214,13 @@ OCA.Sharing.App = {
 		if (state === OC.Share.STATE_REJECTED) {
 			method = 'DELETE';
 		}
-
 		var endPoint = isRemote === true ? 'remote_shares/pending/' : 'shares/pending/';
 		var xhr = $.ajax({
 			url: OC.linkToOCS('apps/files_sharing/api/v1') + endPoint + encodeURIComponent(fileId) + '?format=json',
 			contentType: 'application/json',
 			dataType: 'json',
 			type: method,
-			data: JSON.stringify({
-				...(!!shareType && { shareType }),
-			}),
+			data: JSON.stringify((Boolean(shareType) ? { shareType: shareType } : {})),
```

Comment:
Do you need to convert it to boolean? cant you just do shareType ?

Reply:
@soltanireza65 please clarify for us.

Reply:
I'm not sure but I just remember the datatype was not an int, and it was a string
so an empty string, in this case, would result in Truity